GAwesomeBot V5.0

Gilbert142 opened the pull request:
**Please describe the changes this PR makes and why it should be merged:**
This WIP PR merges GAwesomeBot V5.0 into our default Development branch now that we've reached some stability and performence standards thanks to GADriver. It should be merged because it's the best damn thing to happen to GAB since GAB.

**What does this PR do:**
- [x] This PR changes internal functions, modules and/or utilities
  - [x] This PR modifies the Extension API
- [x] This PR modifies commands
  - [x] This PR changes command functions
  - [x] This PR changes metadata for the command, updated in `commands.js`
  - [x] This PR removes and/or adds commands
- [x] This PR modifies Web processing and/or content
  - [x] This PR modifies static/ejs content
  - [x] This PR modifies request processing (endpoint functions)
  - [x] This PR modifies paths to existing resources
- [ ] This PR **only** includes non-code changes, like changes to default configurations, README, typos, etc.

Commit:
:construction: GADriver migration

Fun fact: Doesn't actually work! But can't be bothered to fix it right now. mongodb module is being dumb

Commit:
Linting

Commit:
:sparkling: GuildMemberUpdate status messages

Commit:
Linting

Commit:
:sparkles: Official mode

Commit:
:bug: Bug Fixes

wasdennnoch committed:
Add roleinfo command (#394)

* Initial roleinfo progress

* Extract roles per page into variable

* Roleinfo changes

* Some more changes half a year later

* Something changed that I have to commit now to mess with git

* Finishing touches

* You won't belive what this commit does!

* This commit will blow your socks off!

Commit:
:bug: Didn't know this was even a possibility...

Commit:
:bug: Fix bug

Commit:
:bug: svr.owner cannot be assumed as defined

Commit:
:rotating_light: Remove linting errors + start linting public JS source files

Commit:
:rocket: :package: Minify source files

Build: c371c928fe8e415206b7dee04255196e24ed1c7f (1899)

Commit:
:bug: RSS Feeds bug fix

Commit:
:rotating_light: Linting

Commit:
:bug: MOTD Bug fix

Commit:
:sparkles: Joke command (#396)

**Please describe the changes this PR makes and why it should be merged:**
Ports over V4 (Dad) Joke command to indev V5, with a new, working API.

**What does this PR do:**
- [ ] This PR changes internal functions, modules and/or utilities
  - [ ] This PR modifies the Extension API
- [x] This PR modifies commands
  - [ ] This PR changes command functions
  - [ ] This PR changes metadata for the command, updated in `commands.js`
  - [x] This PR removes and/or adds commands
- [ ] This PR modifies Web processing and/or content
  - [ ] This PR modifies static/ejs content
  - [ ] This PR modifies request processing (endpoint functions)
  - [ ] This PR modifies paths to existing resources
- [ ] This PR **only** includes non-code changes, like changes to default configurations, README, typos, etc.

Commit:
:bug: Fix some prompt bugs

Commit:
:rocket: :package: Minify source files

Build: f6fdb3b18379a62d4d2626a7b1c16811ec8ec074 (1915)

GilbertGobbels commented:
Also we're gonna rebase, not squash

Commit:
:sparkles: Numfact command (#398)

Commit:
:sparkles: RoleDelete event

Commit:
:speech_balloon: Tweak trivia flavor text + command usage

Commit:
:sparkles: Guild Update event

Commit:
:recycle: Remove duplication

Commit:
:sparkles: Custom Imgur API Key

Commit:
:sparkles: remindme command

Commit:
room & remindme commands (#399)

* :sparkles: remindme command

* :sparkles: room command

* :art: Fix code style issues

* :art: Forgot these

* Add versioning API endpoint to constants

Commit:
Merge remote-tracking branch 'origin/indev-4.0.2' into indev-4.0.2

# Conflicts:
#	Commands/Public/remindme.js

Commit:
:sparkles: say command

Commit:
:bug: Fix Query.pull bug

Commit:
Fix language issue

Commit:
Tweak console file path

Commit:
:bug: Prevent message translation from triggering upon command messages

Commit:
:sparkles: MesageDelete event

Commit:
:sparkles: Nick command (#400)

* :sparkles: Nick command

* :art: Satisfy review requests

Commit:
Indev 4.0.2 shorten (#401)

* :sparkles: remindme command

* :sparkles: shorten command

* Fix linting

wasdennnoch committed:
Rework paginated embed (#283)

* Rework paginated embed, now supports paginating every embed property

* Use empty string as default in format arguments

* Update commands to match the new PaginatedEmbed format

* Small fixes and improvements

* Please the King one commit at a time

Commit:
Merge branch 'development' into indev-4.0.2

Commit:
Delete old eslintrc file

Commit:
:bug: Fix calc bug

Commit:
:sparkles: Allow maintainers to reload events using the reload command

Commit:
:sparkles: Quiet command

Commit:
:bug: RIP GAwesomeBo ... t

Commit:
:bug: Fix ban command deleting message

Commit:
:bug: Fix styling issue

Commit:
:bug: Fix missing userQueryDocument bug

userQueryDocument wasn't properly passed to command functions; fixed!

dependabot[bot] committed:
Bump compression from 1.7.2 to 1.7.3 (#368)

Bumps [compression](https://github.com/expressjs/compression) from 1.7.2 to 1.7.3.
<details>
<summary>Release notes</summary>

*Sourced from [compression's releases](https://github.com/expressjs/compression/releases).*

> ## 1.7.3
>   * deps: accepts@~1.3.5
>     - deps: mime-types@~2.1.18
>   * deps: compressible@~2.0.14
>     - Mark all XML-derived types as compressible
>     - deps: mime-db@'>= 1.34.0 < 2'
>   * deps: safe-buffer@5.1.2
</details>
<details>
<summary>Changelog</summary>

*Sourced from [compression's changelog](https://github.com/expressjs/compression/blob/master/HISTORY.md).*

> 1.7.3 / 2018-07-15
> ==================
> 
>   * deps: accepts@~1.3.5
>     - deps: mime-types@~2.1.18
>   * deps: compressible@~2.0.14
>     - Mark all XML-derived types as compressible
>     - deps: mime-db@'>= 1.34.0 < 2'
>   * deps: safe-buffer@5.1.2
</details>
<details>
<summary>Commits</summary>

- [`becc1c0`](https://github.com/expressjs/compression/commit/becc1c0afb04e0fca5fd12a764b9c9ac8f17a378) 1.7.3
- [`2742fe9`](https://github.com/expressjs/compression/commit/2742fe90e5031e5c2378547b11f937fd684e3835) docs: fix threshold documentation to match implementation
- [`fd7a4d4`](https://github.com/expressjs/compression/commit/fd7a4d47cdbb98941854239f89df88f2d0dea7a3) deps: safe-buffer@5.1.2
- [`5bca5aa`](https://github.com/expressjs/compression/commit/5bca5aaa3c085c0747987c7a2381aaa7d20e8c37) deps: compressible@~2.0.14
- [`c43c578`](https://github.com/expressjs/compression/commit/c43c578e62ed3dbe270089ac84dc088a7f494cd0) build: eslint-plugin-standard@3.1.0
- [`6160662`](https://github.com/expressjs/compression/commit/6160662382513cea9c74fc82d6ea110b2f9af6c9) build: eslint-plugin-promise@3.8.0
- [`d3c8269`](https://github.com/expressjs/compression/commit/d3c8269173c1b17b4a3ff6ad83bbe1aa29aa3993) build: eslint-plugin-import@2.13.0
- [`7ebf291`](https://github.com/expressjs/compression/commit/7ebf291087d69156df4b086e2020e660a1dffb17) build: eslint-plugin-import@2.11.0
- [`b3f7aa5`](https://github.com/expressjs/compression/commit/b3f7aa56066f55c541acf8ab0f8d0b64dc928063) build: eslint@4.19.1
- [`363cbf6`](https://github.com/expressjs/compression/commit/363cbf6fbacd104583273e231919e1b16938cf01) build: Node.js@9.11
- Additional commits viewable in [compare view](https://github.com/expressjs/compression/compare/1.7.2...1.7.3)
</details>
<br />

[![Dependabot compatibility score](https://api.dependabot.com/badges/compatibility_score?dependency-name=compression&package-manager=npm_and_yarn&previous-version=1.7.2&new-version=1.7.3)](https://dependabot.com/compatibility-score.html?dependency-name=compression&package-manager=npm_and_yarn&previous-version=1.7.2&new-version=1.7.3)

Dependabot will resolve any conflicts with this PR as long as you don't alter it yourself. You can also trigger a rebase manually by commenting `@dependabot rebase`.

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot ignore this [patch|minor|major] version` will close this PR and stop Dependabot creating any more for this minor/major version (unless you reopen the PR or upgrade to it yourself)
- `@dependabot ignore this dependency` will close this PR and stop Dependabot creating any more for this dependency (unless you reopen the PR or upgrade to it yourself)
- `@dependabot use these labels` will set the current labels as the default for future PRs for this repo and language
- `@dependabot use these reviewers` will set the current reviewers as the default for future PRs for this repo and language
- `@dependabot use these assignees` will set the current assignees as the default for future PRs for this repo and language
- `@dependabot badge me` will comment on this PR with code to add a "Dependabot enabled" badge to your readme

Additionally, you can set the following in your Dependabot [dashboard](https://app.dependabot.com):
- Update frequency (including time of day and day of week)
- Automerge options (never/patch/minor, and dev/runtime dependencies)
- Pull request limits (per update run and/or open at any time)
- Out-of-range updates (receive only lockfile updates, if desired)
- Security updates (receive only security updates, if desired)

Finally, you can contact us by mentioning @dependabot.

</details>

dependabot[bot] committed:
Bump moment-timezone from 0.5.21 to 0.5.23 (#395)

Bumps [moment-timezone](https://github.com/moment/moment-timezone) from 0.5.21 to 0.5.23.
- [Release notes](https://github.com/moment/moment-timezone/releases)
- [Changelog](https://github.com/moment/moment-timezone/blob/develop/changelog.md)
- [Commits](https://github.com/moment/moment-timezone/commits/0.5.23)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot[bot] committed:
Bump mathjs from 5.0.2 to 5.4.0 (#405)

Bumps [mathjs](https://github.com/josdejong/mathjs) from 5.0.2 to 5.4.0.
<details>
<summary>Changelog</summary>

*Sourced from [mathjs's changelog](https://github.com/josdejong/mathjs/blob/develop/HISTORY.md).*

> # 2018-12-09, version 5.4.0
> 
> - Extended sum.js to accept a dimension input to calculate the sum over a 
>   specific axis. Thanks [**bnlcas**](https://github.com/bnlcas).
> - Fix [#1328](https://github-redirect.dependabot.com/josdejong/mathjs/issues/1328): objects can't be written multi-line. Thanks [**GHolk**](https://github.com/GHolk).
> - Remove side effects caused by `Unit.format` and `Unit.toString`, 
>   making changes to the unit on execution. Thanks [**ericman314**](https://github.com/ericman314).
> 
> 
> # 2018-12-03, version 5.3.1
> 
> - Fixed [#1336](https://github-redirect.dependabot.com/josdejong/mathjs/issues/1336): Unit.toSI() returning units with prefix like `mm` instead 
>   of `m`. Thanks [**ericman314**](https://github.com/ericman314).
> 
> 
> # 2018-11-29, version 5.3.0
> 
> - Implemented function `hasNumericValue`. Thanks [**Sathish-kumar-Subramani**](https://github.com/Sathish-kumar-Subramani).
> - Fix [#1326](https://github-redirect.dependabot.com/josdejong/mathjs/issues/1326): non-ascii character in print.js.
> - Fix [#1337](https://github-redirect.dependabot.com/josdejong/mathjs/issues/1337): `math.format` not working correctly with `{ precision: 0 }`.
>   Thanks [**dkenul**](https://github.com/dkenul).
> 
> 
> # 2018-10-30, version 5.2.3
> 
> - Fixed [#1293](https://github-redirect.dependabot.com/josdejong/mathjs/issues/1293): non-unicode characters in `escape-latex` giving issues in some
>   specific cases. Thanks [**dangmai**](https://github.com/dangmai).
> - Fixed incorrect LaTeX output of function `bitNot`, see [#1299](https://github-redirect.dependabot.com/josdejong/mathjs/issues/1299). Thanks [**FSMaxB**](https://github.com/FSMaxB).
> - Fixed [#1304](https://github-redirect.dependabot.com/josdejong/mathjs/issues/1304): function `pow` not supporting inputs `pow(Unit, BigNumber)`.
> - Upgraded dependencies (`escape-latex@1.2.0`)
> 
> 
> # 2018-10-23, version 5.2.2
> 
> - Fixed [#1286](https://github-redirect.dependabot.com/josdejong/mathjs/issues/1286): Fixed unit base recognition and formatting for
>   user-defined units. Thanks [**ericman314**](https://github.com/ericman314).
> 
> 
> # 2018-10-18, version 5.2.1
> 
> - Fixed unit `rod` being defined as `5.02921` instead of `5.0292`.
>   Thanks [**ericman314**](https://github.com/ericman314).
> - Upgraded dependencies (`fraction.js@4.0.10`)
> - Upgraded devDependencies (`@babel/core@7.1.2`, `nyc@13.1.0`,
>   `webpack@4.21.0`).
> 
> 
> # 2018-10-05, version 5.2.0
> 
> - Implemented support for chained conditionals like `10 < x <= 50`.
></table> ... (truncated)
</details>
<details>
<summary>Commits</summary>

- [`088f845`](https://github.com/josdejong/mathjs/commit/088f84521cd3d9a9fe0711c81bb3a46c9630d19d) Publish v5.4.0
- [`f429bd8`](https://github.com/josdejong/mathjs/commit/f429bd8774a59fc60c96c31919200e797f86546f) Updated history
- [`e45c880`](https://github.com/josdejong/mathjs/commit/e45c8805b3e31977bb31045244160f027077fac8) Add axis for sum request 1244 ([#1310](https://github-redirect.dependabot.com/josdejong/mathjs/issues/1310))
- [`26d396b`](https://github.com/josdejong/mathjs/commit/26d396bcccfc5725e966b8293202ed9d68dc6f29) Update history
- [`2032a9d`](https://github.com/josdejong/mathjs/commit/2032a9d1268c9900a3cb12e02079f50f49a607c2) Fix object literal enable multiple lines ([#1328](https://github-redirect.dependabot.com/josdejong/mathjs/issues/1328)) ([#1349](https://github-redirect.dependabot.com/josdejong/mathjs/issues/1349))
- [`0366c43`](https://github.com/josdejong/mathjs/commit/0366c439e9838b5001fcf67fa3520ec2e954d4ce) Update webpack to the latest version 🚀 ([#1348](https://github-redirect.dependabot.com/josdejong/mathjs/issues/1348))
- [`74b2694`](https://github.com/josdejong/mathjs/commit/74b2694478818560b7860b451e13ca907f748389) Update history
- [`6c03139`](https://github.com/josdejong/mathjs/commit/6c03139ac85038ea13e556f65501145bd29b49ec) Immutable units ([#1344](https://github-redirect.dependabot.com/josdejong/mathjs/issues/1344))
- [`9634f5e`](https://github.com/josdejong/mathjs/commit/9634f5eba4b2670121860531ae33fb3e718fe67d) Update webpack to the latest version 🚀 ([#1347](https://github-redirect.dependabot.com/josdejong/mathjs/issues/1347))
- [`ae1708d`](https://github.com/josdejong/mathjs/commit/ae1708d89c3246b9a8a5b2791a2ceb4ceb1b7bc9) Publish v5.3.1
- Additional commits viewable in [compare view](https://github.com/josdejong/mathjs/compare/v5.0.2...v5.4.0)
</details>
<br />

[![Dependabot compatibility score](https://api.dependabot.com/badges/compatibility_score?dependency-name=mathjs&package-manager=npm_and_yarn&previous-version=5.0.2&new-version=5.4.0)](https://dependabot.com/compatibility-score.html?dependency-name=mathjs&package-manager=npm_and_yarn&previous-version=5.0.2&new-version=5.4.0)

Dependabot will resolve any conflicts with this PR as long as you don't alter it yourself. You can also trigger a rebase manually by commenting `@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot ignore this [patch|minor|major] version` will close this PR and stop Dependabot creating any more for this minor/major version (unless you reopen the PR or upgrade to it yourself)
- `@dependabot ignore this dependency` will close this PR and stop Dependabot creating any more for this dependency (unless you reopen the PR or upgrade to it yourself)
- `@dependabot use these labels` will set the current labels as the default for future PRs for this repo and language
- `@dependabot use these reviewers` will set the current reviewers as the default for future PRs for this repo and language
- `@dependabot use these assignees` will set the current assignees as the default for future PRs for this repo and language
- `@dependabot use this milestone` will set the current milestone as the default for future PRs for this repo and language
- `@dependabot badge me` will comment on this PR with code to add a "Dependabot enabled" badge to your readme

Additionally, you can set the following in your Dependabot [dashboard](https://app.dependabot.com):
- Update frequency (including time of day and day of week)
- Automerge options (never/patch/minor, and dev/runtime dependencies)
- Pull request limits (per update run and/or open at any time)
- Out-of-range updates (receive only lockfile updates, if desired)
- Security updates (receive only security updates, if desired)

Finally, you can contact us by mentioning @dependabot.

</details>

Commit:
Update README's

Commit:
:sparkles: Nuke command (#407)

* :sparkles: Nuke command

* Simplifying nuke command

Commit:
:sparkles: Wolfram command! (#402)

* Refactor fetching data embeds. Need to standardize remaining commands

Commit:
:recycle: Write and implement GABMessage#sendInvalidUsage

Commit:
:sparkles: Reason command (#408)

* :sparkles: Reason command

* PR Requests

Commit:
:bug: Fix traffic counting bug

Commit:
:loud_sound: Various log tweaks

* :loud_sound: Add User-Agent header to verbose incoming request log
* :loud_sound: Improve some logging levels (verbose -> debug for errors)
* :loud_sound: Make CommandHandler timer permanent + verbose
* RIP 666 Commits diff with development

dependabot[bot] committed:
[Security] Bump stringstream from 0.0.5 to 0.0.6 (#385)

Bumps [stringstream](https://github.com/mhart/StringStream) from 0.0.5 to 0.0.6. **This update includes security fixes.**
- [Release notes](https://github.com/mhart/StringStream/releases)
- [Commits](https://github.com/mhart/StringStream/compare/v0.0.5...v0.0.6)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot[bot] committed:
Bump winston-daily-rotate-file from 3.2.3 to 3.5.1 (#391)

Bumps [winston-daily-rotate-file](https://github.com/winstonjs/winston-daily-rotate-file) from 3.2.3 to 3.5.1.
- [Release notes](https://github.com/winstonjs/winston-daily-rotate-file/releases)
- [Commits](https://github.com/winstonjs/winston-daily-rotate-file/compare/v3.2.3...v3.5.1)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot[bot] committed:
Bump eslint-plugin-node from 6.0.1 to 8.0.0 (#403)

Bumps [eslint-plugin-node](https://github.com/mysticatea/eslint-plugin-node) from 6.0.1 to 8.0.0.
- [Release notes](https://github.com/mysticatea/eslint-plugin-node/releases)
- [Commits](https://github.com/mysticatea/eslint-plugin-node/compare/v6.0.1...v8.0.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot[bot] committed:
Bump wikijs from 4.8.0 to 4.10.1 (#404)

Bumps [wikijs](https://github.com/dijs/wiki) from 4.8.0 to 4.10.1.
- [Release notes](https://github.com/dijs/wiki/releases)
- [Commits](https://github.com/dijs/wiki/commits)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot[bot] committed:
Bump mpath from 0.4.1 to 0.5.1 (#410)

Bumps [mpath](https://github.com/aheckmann/mpath) from 0.4.1 to 0.5.1.
- [Release notes](https://github.com/aheckmann/mpath/releases)
- [Changelog](https://github.com/aheckmann/mpath/blob/master/History.md)
- [Commits](https://github.com/aheckmann/mpath/compare/0.4.1...0.5.1)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot[bot] committed:
Bump tail from 1.2.4 to 2.0.2 (#411)

Bumps [tail](https://github.com/lucagrulla/node-tail) from 1.2.4 to 2.0.2.
- [Release notes](https://github.com/lucagrulla/node-tail/releases)
- [Commits](https://github.com/lucagrulla/node-tail/compare/1.2.4...2.0.2)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot[bot] committed:
Bump connect-mongo from 2.0.1 to 2.0.3 (#412)

Bumps [connect-mongo](https://github.com/jdesboeufs/connect-mongo) from 2.0.1 to 2.0.3.
- [Release notes](https://github.com/jdesboeufs/connect-mongo/releases)
- [Changelog](https://github.com/jdesboeufs/connect-mongo/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jdesboeufs/connect-mongo/compare/v2.0.1...v2.0.3)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot[bot] committed:
Bump eslint from 5.1.0 to 5.10.0 (#406)

Bumps [eslint](https://github.com/eslint/eslint) from 5.1.0 to 5.10.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v5.1.0...v5.10.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

Commit:
:sparkles: Unban command (#409)

**Please describe the changes this PR makes and why it should be merged:**

**What does this PR do:**
- [ ] This PR changes internal functions, modules and/or utilities
  - [ ] This PR modifies the Extension API
- [x] This PR modifies commands
  - [ ] This PR changes command functions
  - [ ] This PR changes metadata for the command, updated in `commands.js`
  - [x] This PR removes and/or adds commands
- [ ] This PR modifies Web processing and/or content
  - [ ] This PR modifies static/ejs content
  - [ ] This PR modifies request processing (endpoint functions)
  - [ ] This PR modifies paths to existing resources
- [ ] This PR **only** includes non-code changes, like changes to default configurations, README, typos, etc.

Commit:
:rotating_light: Remove mix of spaces and tabs

Commit:
xkcd command (#416)

* :sparkles: XKCD Command

* Fix linting

* :ok_hand: Xkcd command requests

* :speech_balloon: Fix typos
* :art: Remove code dupe

* :ok_hand: Another code dupe

* :speech_balloon: Revert fixing a typo that wasn't actually a typo

* :recycle: Refactor xkcd command

Commit:
:sparkles: ModLog command (#413)

Commit:
:sparkles: Kick command (#414)

* :sparkles: Kick command

* :bug: Send kicked members a DM before the boot hits

Commit:
:sparkles: Points command (#425)

**Please describe the changes this PR makes and why it should be merged:**

**What does this PR do:**
- [ ] This PR changes internal functions, modules and/or utilities
  - [ ] This PR modifies the Extension API
- [x] This PR modifies commands
  - [ ] This PR changes metadata for commands (such as usage), updated in `commands.js`
- [ ] This PR modifies Web processing and/or content
  - [ ] This PR modifies static/ejs content
  - [ ] This PR modifies request processing (controllers & middleware)
  - [ ] This PR modifies paths to existing resources (routes)
- [ ] This PR **only** includes non-code changes, like changes to default configurations, README, typos, etc.

Commit:
:bug: Fix imgur command bug

Commit:
:sparkles: Adspace for officialMode

Commit:
:bug: Fix EJS Inclusion path bug

Commit:
:bug: Fix misplaced Adspace include

Commit:
:bug: Fix middleware erroring bug

Commit:
:speech_balloon: Update Privacy Policy

Commit:
:bug: Fix server logs not saving timestamp correctly

Commit:
:lipstick: Improve adspace placement

Commit:
:sparkles: Implement an option for users to toggle adspace

Commit:
:rocket: :package: Minify source files

Build: 00c09e873415e3abdca509eb4d06b8878ba9541e (2126)

Commit:
:bug: Fix commandHandler timer for DM's

Commit:
:ambulance: Hotfix outdated RSS module

Commit:
:ambulance: Add new RSS Parser module to package.json

Commit:
:bug: Fix another rss-parser bug

Commit:
:sparkles: Translate command (#426)

* :sparkles: Translate command

* :recycle: Remove callback hell

* :bug: Fix adspace bug

Commit:
:bug: Ignore SYSTEM messages for the default MessageCreate handler

Commit:
:sparkles: messageUpdate event

Commit:
:sparkles: :lipstick: :bug: Replace standard checkboxes with the generic-feature component

* :sparkles: Write .generic-feature, its sub-classes and the javascript that comes with it
* :lipstick: Implement generic-feature on message-of-the-day, moderation and name-display (more to come!)
* :bug: Fix ValidationError not recognizing ValidationErrors

Commit:
Sync lock files

Commit:
:bug: Fix name display typo

Commit:
:lipstick: generic-feature for public-data

Commit:
:art: Format maintainer.ejs correctly

Commit:
:sparkles: Safebooru command!

Commit:
:construction: :lipstick: Beginnings of the extension installer, candidate 2

Commit:
:bug: Fix mention filter page bug

Commit:
:art: Simplify code

Commit:
:construction: :sparkles: :lipstick: Extension installer UI, scopes, extension version control

Commit:
Indev 4.0.2 strike (#427)

* :sparkles: Strike command

* :bug: Allow the master to be ran with a --inspect argument without crashing the shards's worker processes.

* :sparkles: Strikes command!

* :speech_balloon: GAB can do anything

Commit:
:sparkles: Stats command! (#428)

Commit:
:sparkles: Voicetext now works!

* :sparkles: Ported voicetext features from V4 to V5
* :sparkles: Created VoiceStateUpdate event handler
* Update package-lock.json

Commit:
:bug: Room bug fixes

* :bug: Remove room data on channel deletion
* :bug: Remove double argument splicing

Commit:
:rocket: :package: Minify source files

Build: 5943787330143f3a000a6ed06058bcdbc6f65a19 (2221)

dependabot[bot] committed:
Bump vm2 from 3.6.2 to 3.6.4 (#420)

Bumps [vm2](https://github.com/patriksimek/vm2) from 3.6.2 to 3.6.4.
- [Release notes](https://github.com/patriksimek/vm2/releases)
- [Changelog](https://github.com/patriksimek/vm2/blob/master/CHANGELOG.md)
- [Commits](https://github.com/patriksimek/vm2/commits)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot[bot] committed:
Bump moment from 2.22.2 to 2.23.0 (#423)

Bumps [moment](https://github.com/moment/moment) from 2.22.2 to 2.23.0.
- [Release notes](https://github.com/moment/moment/releases)
- [Changelog](https://github.com/moment/moment/blob/develop/CHANGELOG.md)
- [Commits](https://github.com/moment/moment/compare/2.22.2...2.23.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot[bot] committed:
Bump clean-css-cli from 4.1.11 to 4.2.1 (#424)

Bumps [clean-css-cli](https://github.com/jakubpawlowicz/clean-css-cli) from 4.1.11 to 4.2.1.
- [Release notes](https://github.com/jakubpawlowicz/clean-css-cli/releases)
- [Changelog](https://github.com/jakubpawlowicz/clean-css-cli/blob/v4.2.1/History.md)
- [Commits](https://github.com/jakubpawlowicz/clean-css-cli/compare/v4.1.11...v4.2.1)

Signed-off-by: dependabot[bot] <support@dependabot.com>

Commit:
:sparkles: list command (#429)

* :sparkles: list command

* :ok_hand: Review requests!

* :art: Return generateID result
* :art: Move action.join to directly after definition

Commit:
:sparkles: Implement softban in ban command

Commit:
:loud_sound: Log non-ok responses from GAB

Commit:
:rocket: :package: Minify source files

Build: 9f03e0295a97f8ae71e9282a5181e2d0146b5ff4 (2250)

Commit:
:loud_sound: Log validationErrors catched by saveAdminConsoleOptions

Commit:
:bug: Fix validationError bug

Commit:
:bug: Do not edit previous messages for translation messages

Commit:
:bug: Fix master.js eval SyntaxError

Commit:
:construction: Some WIP Extension API stuff

Commit:
:bug: Subtract upvote points from author on mention votes

Commit:
:bug: Fix lottery bugs

Commit:
:sparkles: :lipstick: Extension installer back-end processing + finishing touches!

Extension installer is finished now.

Commit:
:sparkles: :lipstick: :construction: WIP Admin Console extension page

zorbyte committed:
:lipstick: Where are you going Discord? (#435)

Fix: Discord logo was hovering above the "The Best...Bot".

Commit:
:bug: Fix another MOTD bug

They just never end...

Commit:
:bug: Using the channel ID in the error message that the channel doesn't exist is so smart

Commit:
:sparkles: Mute & Unmute commands! (#431)

* :sparkles: Mute command

* :sparkles: Unmute command!

Commit:
:ambulance: Fix major GADriver bug

Javascript memory management is very interesting.

Commit:
:sparkles: :lipstick: :bug: Finish extension web content

All web stuff is done now, which leaves the extension Code API left to go!

Commit:
:rocket: :package: Minify source files

Build: a5a437ab52c1a306e685bcce38498aca0186a0ee (2290)

Commit:
:sparkles: :lipstick: :bug: Introduce script injection

* :sparkles: Script Injection, the friendly kind. Allow maintainers to save JS that is then injected into head.ejs
* :lipstick: Some various CSS tweaks
* :lipstick: Visiting the extension queue now warns you the dangers looming ahead
* :bug: Fix some bugs

Commit:
:rocket: :package: Minify source files

Build: 9508ae5bdcf8ca3d312b71c371504db7eb419210 (2295)

wasdennnoch committed:
Add support for Urban Dictionary links within definitions (#436)

wasdennnoch committed:
Fix formatting character escaping in urban command (#437)

Commit:
:pencil2: Fix typo in stats

Commit:
:construction: WIP Extension Manager + package-lock update

Commit:
:sparkles: Introduce protection against AFK abuse.

* :sparkles: If a User has an AFK message configured, it will be reset once they post a message. This works globally, but not cross-server. For example, posting a message in Server A will remove your Global AFK Message, along with your AFK Message for Server A; your AFK Message for Server B will not be reset.
* :sparkles: AFK messages are not posted in channels or servers where the AFK command is disabled.

Commit:
:sparkles: Disable AFK messages for blocked members.

Commit:
:sparkles: RSS Command

Commit:
:sparkles: :lipstick: Add (Extension) Worker status to maintainer shards page

Commit:
:construction: :sparkles: ExtensionManager, EventsHandler, initial Extensions Worker code

Commit:
:sparkles: Twitter command

Commit:
:sparkles: PresenceUpdate event

Commit:
:fire: We no longer respond to simple presence updates

Commit:
:sparkles: Voice Stats!

Commit:
:sparkles: Auto-removal of empty Voice Rooms.

Commit:
:construction: :sparkles: :recycle: WHAT? We can run extensions!?

* :sparkles: Message (command/keyword) extensions are now executed! No API though, so good luck doing... anything.
* :recycle: Message extensions are triggered by the Shard, using IPC to Worker. This is to save a lot of rewriting and dumb stuff.
* :sparkles: Because this makes extensions a bit more latent, there's a handy loading message while the extension is being executed.
* :sparkles: ObjectID is pretty annoying to work with, so we wrapped it up all nicely in Model.findOneByObjectID

Commit:
:sparkles: userUpdate event handlers!

Never give up on your dreams. If they say that event won't cut it they're wrong, don't listen to the event haters!

Commit:
:sparkles: Restart Extension Code API (from now on referenced to as ECA 'cause I'm lazy) and implement Message

* :fire: :recycle: Make ScopeManager's methods static and refactor the API Scope system.

Commit:
:bug: Don't overwrite existing logs!

Commit:
:sparkles: Show error message when Extension throws an exception

* :sparkles: Execute all Extension code in an async function. Hooray for await!

Commit:
:bug: Various bug fixes

vladfrangu committed:
Update Commands/Public/rss.js

Co-Authored-By: GilbertGobbels <gijsh55@gmail.com>

dependabot[bot] committed:
:arrow_up: Bump express from 4.16.3 to 4.16.4 (#433)

Bumps [express](https://github.com/expressjs/express) from 4.16.3 to 4.16.4.
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/master/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.16.3...4.16.4)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot[bot] committed:
Bump urban from 0.3.1 to 0.3.2 (#417)

Bumps [urban](https://github.com/mvrilo/urban) from 0.3.1 to 0.3.2.
- [Release notes](https://github.com/mvrilo/urban/releases)
- [Commits](https://github.com/mvrilo/urban/commits)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot[bot] committed:
Bump object-sizeof from 1.2.0 to 1.3.0 (#419)

Bumps [object-sizeof](https://github.com/miktam/sizeof) from 1.2.0 to 1.3.0.
- [Release notes](https://github.com/miktam/sizeof/releases)
- [Commits](https://github.com/miktam/sizeof/commits)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot[bot] committed:
:arrow_up: Bump eslint from 5.10.0 to 5.11.1 (#434)

Bumps [eslint](https://github.com/eslint/eslint) from 5.10.0 to 5.11.1.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v5.10.0...v5.11.1)

Signed-off-by: dependabot[bot] <support@dependabot.com>

wasdennnoch committed:
Change config files to config template files (#438)

* Change config files to config template files

So that git can merge new changes to these files without merge conflicts and sad git noobs like me

* yessir

* Update some descriptions

Commit:
:construction: WIP Mystery updater stuff.

thehydrogen committed:
📝 Fix typo in config.js (#464)

* 📝 Fix typo in config.js

* 📑 Fix another typo in config.js

Commit:
:sparkles: Enforce manual configuration for all extensions

Commit:
:ambulance: Some hotfixes.

* :bug: Message count can't be negative, that's bonkers!
* :bug: Sanity check so the activity page doesn't go coco loco.

Commit:
:construction: :sparkles: More Versioning WIP code! Hopefully less unstable now?

* :recycle: Refactor old Updater.js to GAwesomeClient and properly implement everywhere.
* :sparkles: Version downloader fully operationable and working! Hell yeah!
* :recycle: :sparkles: Temp.js is a module used for managing Temp directories. Cool.
* :construction: :sparkles: The start of the version installer is here aswell. More work to be done. A lot of work. A lot, a lot.

Commit:
:ambulance: Fix potential security concern

Commit:
:bug: Fix voiceStateUpdate oopsy.

Commit:
:heavy_plus_sign: Adding security concern dependency

Commit:
:bug: Allow wiki articles to use HTML ID's

Commit:
:bug: Catch everything in READY. No more missing perm errors?

Commit:
:ambulance: Remove usage of deprecated GroupDMChannel

This hopefully fixes self-hosting instances.

Commit:
:bug: Implement new D.js PresenceUpdate event

Commit:
:sparkles: Streamers command + eval command enhances (#440)

* :sparkles: Streamers command + eval command enhances

* :sparkles: Port streamers command over from V4
* :sparkles: If the eval command is executed with the --unsafe argument, tokens will be accessible by the eval function. Tokens are always filtered in the final message sent to Discord, this argument does not affect that behavior.
* :bug: Do not send streamer alerts if no channel is configured

* OWO

Co-Authored-By: GilbertGobbels <gijsh55@gmail.com>

* UWU

Co-Authored-By: GilbertGobbels <gijsh55@gmail.com>

dependabot[bot] committed:
Bump canvas-prebuilt from 2.0.0-alpha.12 to 2.0.0-alpha.14 (#418)

Bumps [canvas-prebuilt](https://github.com/chearon/node-canvas-prebuilt) from 2.0.0-alpha.12 to 2.0.0-alpha.14.
- [Release notes](https://github.com/chearon/node-canvas-prebuilt/releases)
- [Commits](https://github.com/chearon/node-canvas-prebuilt/compare/v2.0.0-alpha.12...v2.0.0-alpha.14)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot[bot] committed:
Bump socket.io from 2.1.1 to 2.2.0 (#421)

Bumps [socket.io](https://github.com/socketio/socket.io) from 2.1.1 to 2.2.0.
- [Release notes](https://github.com/socketio/socket.io/releases)
- [Commits](https://github.com/socketio/socket.io/compare/2.1.1...2.2.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

thehydrogen committed:
✨ Use figlet for bootup message (#446)

* ✨ Change ascii to use figlet

Instead of using a manually made ASCII text on bootup, `figlet` is used.

* 📝 Add figlet to package.json

* 📝 Add comma

dependabot[bot] committed:
:arrow_up: Bump showdown from 1.8.6 to 1.9.0 (#443)

Bumps [showdown](https://github.com/showdownjs/showdown) from 1.8.6 to 1.9.0.
- [Release notes](https://github.com/showdownjs/showdown/releases)
- [Changelog](https://github.com/showdownjs/showdown/blob/master/CHANGELOG.md)
- [Commits](https://github.com/showdownjs/showdown/compare/1.8.6...1.9.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot[bot] committed:
:arrow_up: :lock: Bump extend from 3.0.1 to 3.0.2 (#447)

Bumps [extend](https://github.com/justmoon/node-extend) from 3.0.1 to 3.0.2. **This update includes security fixes.**
- [Release notes](https://github.com/justmoon/node-extend/releases)
- [Changelog](https://github.com/justmoon/node-extend/blob/master/CHANGELOG.md)
- [Commits](https://github.com/justmoon/node-extend/compare/v3.0.1...v3.0.2)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot[bot] committed:
:arrow_up: :lock: Bump lodash from 4.17.10 to 4.17.11 (#448)

Bumps [lodash](https://github.com/lodash/lodash) from 4.17.10 to 4.17.11. **This update includes security fixes.**
- [Release notes](https://github.com/lodash/lodash/releases)
- [Changelog](https://github.com/lodash/lodash/blob/master/CHANGELOG)
- [Commits](https://github.com/lodash/lodash/compare/4.17.10...4.17.11)

Signed-off-by: dependabot[bot] <support@dependabot.com>

Commit:
:rotating_light: Fix linting errors in master.js

Commit:
:sparkles: Youtube command

Commit:
:fire: With Discord's mobile app now featuring fully fledged permission editing, the perms command is obsolete and will not be updated to V5.

Commit:
:fire: Remove traces of the perms command

wasdennnoch committed:
Add `content` support to PaginatedEmbed

And rename embedTemplate to messageTemplate as it's not strictly holding embed data anymore.

Commit:
:sparkles: Weather command

Commit:
:sparkles: Time command

Commit:
:bug: Fix sharding miscalculation bug + activity page error

Commit:
:sparkles: Show error message when using the disabled shorten command

Commit:
:sparkles: Introduce GDA ranking.

Commit:
:bug: Fix PresenceUpdate bugs + RSS Timeouts

* :bug: Set RSS Feed parser timeout to 2 seconds

Commit:
:bug: More presenceupdate bugs.

* What a messy file...

Commit:
:loud_sound: Improve GADriver Query error logging

Commit:
:construction: Various stability Fixes

Commit:
:bug: PresenceUpdate bug fix

Commit:
:sparkles: :bug: :art: :package: Update winston, rewrite Console to Logger, catch more promises

* :sparkles: :art: Implement Logger: a wrapping class for winston 3 with better logging!
* :package: Update winston to winston 3
* :bug: Catch many previously uncaught Discord API functions.

Commit:
:mute: Downgrade ignoring logs to verbose.

Commit:
:bug: Fix extension queue gallery for guests

Commit:
:bug: Ignore non-guild voiceStateUpdates

Commit:
:sparkles: Sentry integration for Logger

* :recycle: Move OfficialMode/Prod ID's to Constants, in anticipation of refactoring "OfficialMode" to stages.

Commit:
:sparkles: Tag process type in Sentry events

Commit:
:bug: Alias "warn" to "warning" for Sentry

Commit:
:bug: Fix outstanding bugs

* :bug: Text.COMMAND_ERR -> Text.ERROR_TITLE.
* :bug: Clear timeout for messageListeners correctly.
* :bug: Re-implement sharding for Shard workers correctly.
* :bug: Fix debug log for HelpMenu.
* :bug: Catch 404 DiscordAPIErrors in GABMessage._handleSendError
* :bug: Validate the existence of the required channelDocument when auto-ending a Lottery.
* :bug: Fix controllers.dashboard.stats.post by validating body payload.

Commit:
:construction: :recycle: Add patch verification to the updater.

* :recycle: Refactor Logger's Sentry error capturing to Logger.sendSentryError

Commit:
:rocket: :package: Minify source files

Build: 38b06de501f663a8cc0bc2987912e94cc9cf7bde (2875)

Commit:
:bug: Fix maintainer console logs section for new Logger

Commit:
:sparkles: Enhance the extension experience

* :sparkles: Add an in-browser source code viewer.
* :sparkles: Show the current review status of local extensions with more accuracy
* :sparkles: Label local extensions in the gallery.

Commit:
:rotating_light: Fix linting errors

Commit:
:rocket: :package: Minify source files

Build: 9496b76ae3a2088dd7d7940baecc837ea4e8d7a8 (2885)

Commit:
:bug: Fix ClearServerStats bug

Depend on memberDocument._id rather than member.id

Commit:
:bug: Specify version for Source viewer to fix Queue extensions

Commit:
:rocket: :package: Minify source files

Build: c8928417965c5569f8d3e99337433579f5c32225 (2891)

Commit:
:bug: Fix extensions path

Commit:
:bug: Fix all relative extension paths

Commit:
:bug: Limit the servers API endpoint correctly

vladfrangu commented on .eslintrc.js:
```diff
@@ -1,34 +1,39 @@
+		es6: true,
+	},
+	parserOptions: {
+		ecmaVersion: 2018,
```
```suggestion
		ecmaVersion: 2019,
```

Commit:
:bug: :lock: Ignore unaccepted extension versions and disallow the execution of them.

* :bug: Fix debugDump

vladfrangu commented on Commands/PM/giveaway.js:
```diff
@@ -8,8 +8,8 @@ module.exports = async ({ bot, Constants: { Colors } }, msg, commandData) => {
-						name: bot.user.username,
-						icon_url: bot.user.avatarURL(),
+						name: client.user.username,
+						icon_url: client.user.avatarURL(),
```
can we change all these to displayAvatarURL thaanks

Commit:
:rocket: :package: Minify source files

Build: 9f4c94b1cb3c84a930a15198d2aa272d85a1880f (2899)

Commit:
:bookmark: Bump to v5.0.0

vladfrangu commented on Commands/Public/ban.js:
```diff
@@ -83,76 +89,92 @@ module.exports = async ({ client, Constants: { Colors, Text }, configJS }, { ser
+						} else {
+							msg.guild.members.ban(member.id, { days, reason: `${reason} | Command issued by @${msg.author.tag}` });
+						}
+						dmBanned(member.id);
```
We should probably try this BEFORE banning, in case its the only guild a user shares with gab

Gilbert142 replied:
:+1: While we're at it, let's catch those `ban` Promises.

vladfrangu commented on Commands/Public/countdown.js:
```diff
@@ -0,0 +1,99 @@
+
+module.exports = async ({ client, Constants: { Colors, Text } }, { serverDocument, serverQueryDocument }, msg, commandData) => {
+	if (msg.suffix) {
+		if (msg.suffix.includes("|") || msg.suffix.indexOf(" in ") > -1) {
```
```suggestion
		if (msg.suffix.includes("|") || msg.suffix.includes(" in ")) {
```

vladfrangu commented on Commands/Public/dogfact.js:
```diff
@@ -0,0 +1,35 @@
+	const { body, statusCode, statusText } = await get(APIs.DOGFACT(number));
+	if (statusCode === 200 && body && body.facts.length) {
+		const descriptions = [];
+		body.facts.forEach(d => {
```
God forgive me for this sin. Simplify this to a map

vladfrangu commented on Commands/Public/expand.js:
```diff
@@ -0,0 +1,87 @@
+	PHISHTANK: `Phishtank`,
+};
+
+module.exports = async ({ Constants: { Colors, Text, APIs } }, documents, msg, commandData) => {
```
spoopy.link is dead :f:

Gilbert142 replied:
Do you know any alternatives?

vladfrangu commented on Commands/Public/info.js:
```diff
@@ -0,0 +1,107 @@
+	if (guild.features.includes("VANITY_URL")) {
+		let customInvite;
+		if (guild.me.permissions.has("MANAGE_GUILD")) {
+			customInvite = (await client.api.guilds(guild.id)["vanity-url"].get()).code;
```
So... guild.vanityCode exists now in latest d.js :) https://discord.js.org/#/docs/main/master/class/Guild?scrollTo=vanityURLCode

vladfrangu commented on Commands/Public/joke.js:
```diff
@@ -0,0 +1,36 @@
+				color: Colors.SOFT_ERR,
+				description: "Failed to fetch a joke...",
+				footer: {
+					text: "Failure is funny too, right?",
```
Too depressing, pls nerf

Gilbert142 replied:
Errors are depressing.

vladfrangu commented on Commands/Public/ping.js:
```diff
@@ -0,0 +1,24 @@
+	msg.send({
+		embed: {
+			color: Colors.LIGHT_GREEN,
+			title: "Pong! 🏓",
```
I mean if we want to be consistent, we can make this: "Received the pong"

Gilbert142 replied:
You must be fun at parties.

vladfrangu commented on Commands/Public/roleinfo.js:
```diff
@@ -0,0 +1,93 @@
+		}
+
+		const memberRoles = [...msg.member.roles.values()].sort((a, b) => b.position - a.position);
+		const totalPermissionsBitfield = memberRoles.reduce((a, b) => a | b.permissions, 0); // eslint-disable-line no-bitwise
```
What is this.. Member.permissions

Gilbert142 replied:
It's not... This is Role.permissions

Check out `msg.member.roles.values()`.

vladfrangu commented on Commands/Public/strike.js:
```diff
@@ -0,0 +1,100 @@
+	if (msg.suffix) {
+		let [member, ...reason] = ArgParser.parseQuoteArgs(msg.suffix, msg.suffix.includes("|") ? "|" : " ");
+		member = await client.memberSearch(member.trim(), msg.guild).catch(() => null);
+		if (reason) reason = reason.join(" ").trim();
```
reason will always be truthy iirc

Gilbert142 replied:
Yeah, `!![]` is `true`. Will change.

vladfrangu commented on Commands/Public/strikes.js:
```diff
@@ -0,0 +1,62 @@
+module.exports = async ({ Constants: { Colors, Text }, client }, { serverDocument, serverQueryDocument }, msg, commandData) => {
+	let member;
+	// eslint-disable-next-line prefer-destructuring
+	if (!msg.suffix || msg.suffix === "me") member = msg.member;
```
```suggestion
	if (!msg.suffix || msg.suffix === "me") ({ member } = msg.member);
```

Gilbert142 replied:
`msg.member.member` 🤔

vladfrangu replied:
Remove the member from msg.member :)

Gilbert142 replied:
RIP github code suggestion committing

vladfrangu reviewed (dismissed):
Can you believe me when I say i legit cant review this entire PR?

GilbertGobbels commented:
> Can you believe me when I say i legit cant review this entire PR?

You're clearly not trying hard enough.

Commit:
:construction: Small Extension API updates

Commit:
:bug: Catch errors when ensuring channelDocuments.

Commit:
:recycle: Edit message instead of deletion + winston -> logger

Commit:
:bug: Catch rss command errors

* Swallow since logging is already implemented on GABMessage.send

dependabot-preview[bot] committed:
:arrow_up: Bump canvas-constructor from `2f94122` to `04134b9`

Bumps [canvas-constructor](https://github.com/kyranet/canvasConstructor) from `2f94122` to `04134b9`.
- [Release notes](https://github.com/kyranet/canvasConstructor/releases)
- [Commits](https://github.com/kyranet/canvasConstructor/compare/2f941220f01844573c2df4869686be9a073509b9...04134b9e8673acf0e06d151731fa4c5fb089a2fc)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

Commit:
:bug: :sparkles: Fix various bugs and reset VoiceStatsCollector on READY

Commit:
:rocket: :package: Minify source files

Build: 5aab1bbb3d44cbd8e7d46abaf6c90dc9207c5856 (2951)

dependabot-preview[bot] committed:
:arrow_up::lock: Bump lodash from 4.17.11 to 4.17.15

Bumps [lodash](https://github.com/lodash/lodash) from 4.17.11 to 4.17.15. **This update includes security fixes.**
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.17.11...4.17.15)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
:arrow_up: Bump mpath from 0.5.1 to 0.6.0

Bumps [mpath](https://github.com/aheckmann/mpath) from 0.5.1 to 0.6.0.
- [Release notes](https://github.com/aheckmann/mpath/releases)
- [Changelog](https://github.com/aheckmann/mpath/blob/master/History.md)
- [Commits](https://github.com/aheckmann/mpath/compare/0.5.1...0.6.0)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
:arrow_up: Bump eslint-plugin-promise from 3.8.0 to 4.2.1

Bumps [eslint-plugin-promise](https://github.com/xjamundx/eslint-plugin-promise) from 3.8.0 to 4.2.1.
- [Release notes](https://github.com/xjamundx/eslint-plugin-promise/releases)
- [Changelog](https://github.com/xjamundx/eslint-plugin-promise/blob/master/CHANGELOG.md)
- [Commits](https://github.com/xjamundx/eslint-plugin-promise/commits)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
:arrow_up: Bump wikijs from 4.13.0 to 6.0.0

Bumps [wikijs](https://github.com/dijs/wiki) from 4.13.0 to 6.0.0.
- [Release notes](https://github.com/dijs/wiki/releases)
- [Commits](https://github.com/dijs/wiki/commits)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
:arrow_up: Bump express-rate-limit from 2.11.0 to 5.0.0

Bumps [express-rate-limit](https://github.com/nfriedly/express-rate-limit) from 2.11.0 to 5.0.0.
- [Release notes](https://github.com/nfriedly/express-rate-limit/releases)
- [Commits](https://github.com/nfriedly/express-rate-limit/compare/v2.11.0...v5.0.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
:arrow_up: Bump jimp from 0.2.28 to 0.6.4

Bumps [jimp](https://github.com/oliver-moran/jimp) from 0.2.28 to 0.6.4.
- [Release notes](https://github.com/oliver-moran/jimp/releases)
- [Commits](https://github.com/oliver-moran/jimp/commits)

Signed-off-by: dependabot[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
:arrow_up: Bump connect-mongo from 2.0.3 to 3.0.0

Bumps [connect-mongo](https://github.com/jdesboeufs/connect-mongo) from 2.0.3 to 3.0.0.
- [Release notes](https://github.com/jdesboeufs/connect-mongo/releases)
- [Changelog](https://github.com/jdesboeufs/connect-mongo/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jdesboeufs/connect-mongo/compare/v2.0.3...v3.0.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
:arrow_up: Bump mathjs from 5.10.3 to 6.1.0

Bumps [mathjs](https://github.com/josdejong/mathjs) from 5.10.3 to 6.1.0.
- [Release notes](https://github.com/josdejong/mathjs/releases)
- [Changelog](https://github.com/josdejong/mathjs/blob/develop/HISTORY.md)
- [Commits](https://github.com/josdejong/mathjs/compare/v5.10.3...v6.1.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
:arrow_up: Bump eslint from 5.16.0 to 6.2.1

Bumps [eslint](https://github.com/eslint/eslint) from 5.16.0 to 6.2.1.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v5.16.0...v6.2.1)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

vladfrangu commented on Commands/Public/ban.js:
```diff
@@ -83,76 +89,92 @@ module.exports = async ({ client, Constants: { Colors, Text }, configJS }, { ser
+				if (message.content) {
+					collector.stop();
+					try {
+						await message.delete();
```
Turn this whole thing into `if (message.deletable) await message.delete();`

If possible to do the same everywhere?

Gilbert142 replied:
How about we extend Message.delete to include this internally?

vladfrangu replied:
Go for it!

vladfrangu commented on Commands/Public/e621.js:
````diff
@@ -0,0 +1,85 @@
+					];
+					if (unparsed[i].description) {
+						const d = `\`\`\`css\n${unparsed[i].description.split("").slice(0, 1500).join("")
+							.replace(/(\[\/?b\]|\[\/?u\]|\[\/?i\])/g, "")}${unparsed[i].description.length > 1500 ? `...\`\`\` Read more [here](https://e621.net/post/show/${unparsed[i].id})` : "```"}`;
````
What the holy jesus is this regex, and why isn't it like `/\[(?:\/?b|u|i)\]/`

vladfrangu commented on Commands/Public/info.js:
```diff
@@ -0,0 +1,107 @@
+	});
+	const numbersText = [
+		`👥 Members: **${guild.memberCount}** (of which **${onlineMembers}** ${onlineMembers === 1 ? "is" : "are"} online)`,
+		`🏷 Roles: **${guild.roles.size - 1}**`,
```
Why... are we subtracting one... That's wrong.

That'd make guilds with just the @ everyone role say `Roles: 0`

Gilbert142 replied:
Is that really wrong? :thinking:

vladfrangu replied:
oh, wait, true. Disregard

vladfrangu commented on Commands/Public/info.js:
```diff
@@ -0,0 +1,107 @@
+	}
+	if (guild.features.includes("VANITY_URL")) {
+		let customInvite;
+		if (guild.me.permissions.has("MANAGE_GUILD")) {
```
No need for this perm check, as we always get this

Gilbert142 replied:
We always get the vanityURL?

vladfrangu replied:
We get the code, yes. The correct way to know if a guild is partnered is checking if features includes "PARTNERED". But we will always get the code, so turn that if check into `if (guild.vanityWhateverItWas) `

TTtie replied:
*(autocomplete)* `guild.vanityURLCode`

vladfrangu commented on Commands/Public/rss.js:
```diff
@@ -0,0 +1,84 @@
+					color: Colors.SOFT_ERR,
+					description: "I couldn't read that RSS Feed 🏳️",
+				},
+			}).catch(() => null);
```
useless

Gilbert142 replied:
Except that it't not useless at all and prevents unhandledRejections?

vladfrangu replied:
That specific one IS useless since the await above would throw if it can't send a message

Gilbert142 replied:
Didn't see the await my bad.

Gilbert142 replied:
Hold up I didn't see the await because there isn't one @vladfrangu

vladfrangu replied:
![image](https://user-images.githubusercontent.com/17960496/63636994-d8aeee80-c67f-11e9-9894-376fb086ea76.png)

vladfrangu commented on Commands/Public/safebooru.js:
````diff
@@ -0,0 +1,74 @@
+		body.forEach(post => {
+			descriptions.push(`Uploaded by ${post.uploader_name}\nClick [here](https://safebooru.donmai.us/posts/${post.id}) to see the full post.${!post.description || post.description === "" ? "" :
+				`\`\`\`css\n${post.description.split("").slice(0, 1500).join("")
+					.replace(/(\[\/?b\]|\[\/?u\]|\[\/?i\])/g, "")}${post.description.length > 1500 ? `...\`\`\` Read more [here](https://safebooru.donmai.us/posts/${post.id})` : "```"}`}`);
````
As before in e621

vladfrangu commented on Commands/Public/strike.js:
```diff
@@ -0,0 +1,99 @@
+			const DMChannel = await member.user.createDM();
+			await DMChannel.send({
+				embed: {
+					description: `Tsching! You just received your ${strikeAmount} strike! ⚡`,
```
While I love this message (gilbert you're a genius), `You just received your 5 strike` sounds... bad..

We need the `th`, `st`, `nd`, `rd`s. Also, bold them!!

Gilbert142 replied:
Thank you 😊 
And yes, you're correct!

Gilbert142 replied:
@vladfrangu How foolish of you to underestimate my coding analytic thought process and perfect commit skills. Check line 48 again and bathe in your own inferiority complex you absolute fool.

vladfrangu replied:
Which doesn't handle the `21st`. FOOL!

vladfrangu commented on Commands/Public/strike.js:
```diff
@@ -0,0 +1,99 @@
+		msg.send({
+			embed: {
+				color: Colors.SUCCESS,
+				description: `Feel the thunder! **@${client.getName(serverDocument, member)}** has received their ${strikeAmount} strike! 🚦`,
```
^

vladfrangu commented on Commands/Public/strike.js:
```diff
@@ -0,0 +1,99 @@
+				color: Colors.SUCCESS,
+				description: `Feel the thunder! **@${client.getName(serverDocument, member)}** has received their ${strikeAmount} strike! 🚦`,
+				footer: {
+					text: success ? "I also warned them via DM ⚠" : "I tried to warn them via DM, but something went wrong!",
```
```suggestion
					text: success ? "I also warned them via DM ⚠" : "I tried to warn them via DM, but they denied my DM..",
```

Gilbert142 replied:
Double usage of DM ticks me off

vladfrangu replied:
"But they denied me?" with a sad face then

vladfrangu commented on Commands/Public/twitter.js:
```diff
@@ -0,0 +1,69 @@
+			num = parseInt(num);
+		}
+
+		const m = await msg.channel.send(Text.THIRD_PARTY_FETCH("The birbs are hard at work 🐦"));
```
```suggestion
		const m = await msg.channel.send(Text.THIRD_PARTY_FETCH("We're looking for the birbs 🐦"));
```

vladfrangu commented on Commands/Public/unban.js:
```diff
@@ -0,0 +1,74 @@
+				embed: {
+					color: Colors.SOFT_ERR,
+					title: `I'm sorry, but I can't do that... 😔`,
+					description: `I'm missing permissions to ban that user!`,
```
```suggestion
					description: `I'm missing permissions to unban that user!`,
```

vladfrangu commented on Commands/Public/unban.js:
```diff
@@ -0,0 +1,74 @@
+		query = query.slice(2, -1);
+	}
+
+	return user.id === query.toLowerCase().trim() || user.tag === query.toLowerCase().trim() || user.username === query.toLowerCase().trim();
```
Useless trims, since you do it when you call it

vladfrangu commented on Commands/Public/unban.js:
```diff
@@ -0,0 +1,74 @@
+				msg.send({
+					embed: {
+						color: Colors.SUCCESS,
+						description: `You've successfully given **@${ban.user.tag}** a second chance 😇`,
```
Add a footer and make it say `They better not blow it again..` I dare you

Gilbert142 replied:
Challenge accepted, I'll do you one better and use 3 dots because I'm not insane.

vladfrangu commented on Commands/Public/unban.js:
```diff
@@ -0,0 +1,74 @@
+					color: Colors.SOFT_ERR,
+					description: "I couldn't find a matching user that's banned on this guild!",
+					footer: {
+						text: isNaN(query) || query.length !== 18 ? "Try unbanning by ID instead." : "",
```
IDs can be between 17 and 19 chars, inclusive (we're still far from 19 digits, but they're coming)

Gilbert142 replied:
Scary

vladfrangu commented on Commands/Public/urban.js:
```diff
@@ -0,0 +1,66 @@
+const { get } = require("snekfetch");
+const PaginatedEmbed = require("../../Modules/MessageUtils/PaginatedEmbed");
+
+module.exports = async ({ Constants: { Colors, Text, APIs } }, documents, msg, commandData) => {
```
Should we... lock urban behind the nsfw wall? Some responses and queries can be very... NSFW

Gilbert142 replied:
Nah it's not actual images.

vladfrangu commented on Commands/Public/weather.js:
```diff
@@ -0,0 +1,53 @@
+
+module.exports = async ({ Constants: { APIs, Colors, Text }, client }, documents, msg, commandData) => {
+	if (!msg.suffix) {
+		await msg.sendInvalidUsage(commandData, "It's always sunny in robot land 🌞", "You must supply a city to lookup!");
```
```suggestion
		await msg.sendInvalidUsage(commandData, "It's always raining bytes in bot land 🌞", "You must supply a city to lookup!");
```

vladfrangu commented on Commands/Public/year.js:
```diff
@@ -0,0 +1,23 @@
+	const days = Math.floor(duration / (1000 * 60 * 60 * 24));
+	return msg.send({
+		embed: {
+			color: Colors.INFO,
```
Maybe we can add a little bit of this?

```suggestion
			color: Colors.INFO,
			title: "Woo! Prepare the party poppers~",
```

Gilbert142 replied:
The `~` at the end makes it a surprisingly large amount more sexual.

vladfrangu commented on Commands/Public/youtube.js:
```diff
@@ -0,0 +1,76 @@
+				color: Colors.SOFT_ERR,
+				description: `No results were found for **${query}**. 😞`,
+			},
+			disableEveryone: true,
```
Useless for embeds, they don't ping

vladfrangu commented on Commands/Shared/eval.js:
```diff
@@ -2,15 +2,19 @@ const { Gist, RegExpMaker } = require("../../Modules/Utils/");
+		const hrstart = process.hrtime();
+		const forceUnsafe = msg.suffix.split(" ")[0].toLowerCase() === "--unsafe";
+		let { suffix } = msg;
+		if (forceUnsafe) suffix = suffix.split(" ").splice(1).join(" ");
```
This will never work. splice returns the removed elements

vladfrangu commented on Internals/Boot.js:
```diff
@@ -0,0 +1,158 @@
+};
+
+const token = (val, configJS, configJSON, auth) => {
+	if (typeof val !== "string" && typeof val !== "number") {
```
That number doesn't seem right

vladfrangu commented on Internals/Events/channelDelete/GAB.ChannelDelete.js:
```diff
@@ -0,0 +1,97 @@
+		// Save possible changes
+		if (updated) {
+			try {
+				await	serverDocument.save();
```
```suggestion
				await serverDocument.save();
```

vladfrangu commented on Internals/Events/channelDelete/GAB.ChannelDelete.js:
```diff
@@ -0,0 +1,97 @@
+		}
+
+		// Command channel configurations
+		for (const command in serverDocument.config.commands) {
```
Sweeet jesus no, please for-of Object.entries

Gilbert142 replied:
Fine, I'll bite. HoW sO?

vladfrangu replied:
```js
for (const [key, command] of Object.entries(serverDocument.config.commands))
```

Gilbert142 replied:
no i know how it works you dingus

vladfrangu commented on Internals/Events/message/GAB.MessageCreate.js:
```diff
@@ -1,575 +1,548 @@
+			return false;
+		}
+		if (msg.type !== "DEFAULT") {
+			logger.verbose(`Ignoring non-standard message.`, { msgid: msg.id, usrid: msg.author.id, chid: msg.channel.id });
```
Except you're not, you goofball

Gilbert142 replied:
This comment made  me laugh out loud

vladfrangu commented on Internals/Events/ready/GAB.Ready.js:
```diff
@@ -1,429 +1,407 @@
+		let leftGuilds = 0;
+		if (this.configJSON.guildBlocklist.length) {
+			this.configJSON.guildBlocklist.forEach(guildID => {
+				if (this.client.guilds.get(guildID)) {
```
```js
const guild = this.client.guilds.get(guildID);
if (guild) {
	yes;
}
```

vladfrangu commented on Internals/Extensions/API/Modules/Extension.js:
```diff
@@ -0,0 +1,68 @@
+		Error: GABError,
+	},
+} = require("../../../index");
+let rawExtensionDocument;
```
Is never set

Gilbert142 replied:
Extensions/API is in-progress, there's no point to reviewing it at this stage.

vladfrangu commented on Modules/Migration.js:
```diff
@@ -1,50 +1,96 @@
-module.exports = () => {
-	winston.info(`Preparing to migrate ${databaseURL} to GAB 4.1`);
+module.exports = async () => {
+	logger.info(`Preparing to migrate database "${database.db}" on "${database.url}" to GADRiver standards...`);
```
```suggestion
	logger.info(`Preparing to migrate database "${database.db}" on "${database.url}" to GADriver standards...`);
```

wasdennnoch replied:
Cry me a GADRiver...

GilbertGobbels replied:
You guys can cry me a GADRiver!

vladfrangu commented on Modules/Utils/RemoveFormatting.js:
```diff
@@ -0,0 +1,13 @@
+		.replace(/_/g, "\\_")
+		.replace(/\*/g, "\\*")
+		.replace(/`/g, "\\`")
+		.replace(/@(everyone|here|\d{17,18}>)/g, "@\u200b$1");
```
This is wrong, as it will produce `@\u200b12345678`

vladfrangu commented on master.js:
```diff
@@ -0,0 +1,356 @@
+			await logger.info(`Connected to MongoDB successfully.`);
+
+			logger.silly("Confirming MongoDB config values.");
+			const adminDatabaseResponse = await db.client.admin().command({ getCmdLineOpts: 1 });
```
This WILL THROW if we cannot access the admin DB. And truth be told, at that point it's safe to assume you are not dumb enough to expose your db

vladfrangu reviewed (changes requested):
Have fun Fish! ❤️ 

![image](https://user-images.githubusercontent.com/17960496/63623268-ce4e0f80-c601-11e9-9a28-52d33f613dd7.png)

Commit:
:ambulance: Fix critical ClearServerStats bug

Client.checkRank is the bane of my existence.

Commit:
:sparkles: :bug: Finish the Version Manager/Updater!

* :bug: Fix various bugs all around.

Commit:
:rotating_light: Ignore no-useless-catch for now

Commit:
:rotating_light: Ignore no-prototype-builtins for now

Commit:
:bug: Fix checkRank incorrect Promise.catch

Commit:
:sparkles: Implement official now-updated versioning server!

Commit:
:bug: Extend GABMessage.delete and catch 403

Avoid making a request to Discord is GABMessage.deletable is false.

Commit:
:bug: Reduce regex complexity for e621 and safebooru

Commit:
:art: Remove unnecessary permission check for the info command

Commit:
:speech_balloon: Various text updates

Commit:
:bug: Fix splice/slice typo in eval

Commit:
:art: Remove unnecessary type check in the token parameter

Commit:
:art: Fix typo tab

Commit:
:bug: Stop lying and actually ignore non-standard messages

Commit:
:bug: Ensure strike counts above 20 are displayed correctly

Commit:
:art: Simplify for...in to for...of in ChannelDelete

Commit:
:zap: Reduce guild lookups when leaving blocklisted guilds

Commit:
:fire: Remove hopefully now obsolete mongodb config warning.

GilbertGobbels commented:
> By the way, I did the regex wrong, move the `\/` before the `()`

I don't think that's right either...

Commit:
:speech_balloon: Fix typo in Migration script

Commit:
:speech_balloon: Third time's the charm

vladfrangu commented:
> I don't think that's right either...

![image](https://user-images.githubusercontent.com/17960496/63637626-00ee1b80-c687-11e9-8fd3-554524c55bfb.png)

GilbertGobbels commented:
> > I don't think that's right either...
> 
> ![image](https://user-images.githubusercontent.com/17960496/63637626-00ee1b80-c687-11e9-8fd3-554524c55bfb.png)

Let's be fair, you didn't mention the question mark.

Commit:
:bug: Fix incorrect regex in naughty commands

Commit:
:bug: :package: Slay the hydra, sync the lock file

vladfrangu commented on Commands/Public/e621.js:
````diff
@@ -0,0 +1,85 @@
+					];
+					if (unparsed[i].description) {
+						const d = `\`\`\`css\n${unparsed[i].description.split("").slice(0, 1500).join("")
+							.replace(/\[\/?(?:b|u|i)\]/g, "")}${unparsed[i].description.length > 1500 ? `...\`\`\` Read more [here](https://e621.net/post/show/${unparsed[i].id})` : "```"}`;
````
How about we remove this `Read more` and add just the click here at the bottom?

Gilbert142 replied:
But... why?

vladfrangu replied:
Duplicate `Click here to see the full post` tbh..

Gilbert142 replied:
I guess
We would miss some kind of indicator that we truncated though.

vladfrangu replied:
We still have the `...` at the end of the text

vladfrangu commented on Commands/Public/info.js:
```diff
@@ -0,0 +1,104 @@
+	if (guild.features.includes("INVITE_SPLASH")) {
+		specialText.push(`**»** This server can use a **custom invite splash background**!${guild.splash ? ` It is currently set to [this](${guild.splashURL({ format: "png", size: 2048 })})` : ""}`);
+	}
+	if (guild.features.includes("VANITY_URL")) {
```
I said turn this into `if (guild.vanityURLCode)`

Gilbert142 replied:
That doesn't make sense with the message

vladfrangu replied:
It does tho? if you have a vanityURLCode, you have a discord.gg URL

Gilbert142 replied:
The message says the server *can* have a Vanity URL. It only mentions the server *actually* having one if vanityURLCode exists.

Commit:
:zap: Remove extremely inefficient string mutation

Commit:
:bug: Do not attempt to handle VoiceStateUpdate if serverDocument is missing

Commit:
:bug: Avoid using Message.member for Status Messages
